hypervisor/dm/vpci/sharing_mode.c

```diff
@@ -106,20 +106,16 @@ static void sharing_mode_cfgwrite(__unused struct vpci *vpci, union pci_bdf bdf,
 
 static struct pci_vdev *alloc_pci_vdev(struct acrn_vm *vm, union pci_bdf bdf)
 {
-	struct pci_vdev *vdev;
-
 	if (num_pci_vdev >= CONFIG_MAX_PCI_DEV_NUM) {
 		return NULL;
 	}
 
-	vdev = &sharing_mode_vdev_array[num_pci_vdev++];
-
 	/* vbdf equals to pbdf otherwise remapped */
-	vdev->vbdf = bdf;
-	vdev->vpci = &vm->vpci;
-	vdev->pdev.bdf = bdf;
+	sharing_mode_vdev_array[num_pci_vdev].vbdf = bdf;
+	sharing_mode_vdev_array[num_pci_vdev].vpci = &vm->vpci;
+	sharing_mode_vdev_array[num_pci_vdev].pdev.bdf = bdf;
 
-	return vdev;
+	return &sharing_mode_vdev_array[num_pci_vdev++];
```

Comment:
I remember MISRAC also requires us to not use X++ together with others in many cases -- please check with MISRAC team. Need to split into 2 senetences.

Reply:
We still have a few new violations in using X++ & ++X.

Reply:
OK, I will check with Junjie and Huihuang.